trace.go

```diff
@@ -185,11 +219,35 @@ type rawTrace struct {
 	Message string `json:"message"`
 }
 
+// AddUserMessage adds user-friendly message describing the error nature
+func (e *TraceErr) AddUserMessage(formatArg interface{}, rest ...interface{}) {
+	newMessage := fmt.Sprintf(fmt.Sprintf("%v", formatArg), rest...)
+	if len(e.Message) == 0 {
+		e.Message = newMessage
+	} else {
+		e.Message = strings.Join([]string{e.Message, newMessage}, ", ")
+	}
+}
+
+// UserMessage returns user-friendly error message
+func (e *TraceErr) UserMessage() string {
+	if e.Message != "" {
+		return e.Message
+	}
+	return e.Err.Error()
+}
+
+// DebugReport returns develeoper-friendly error report
+func (e *TraceErr) DebugReport() string {
```

Comment:
the only reason is to return user friendly message by default, not a debug stack trace in this particular case. Full stack trace is super confusing